src/components/header/signInHeader/signInHeader.js

```diff
@@ -8,38 +9,50 @@ import { Input } from '../../toolbox/inputs';
 import { addHttp, getAutoLogInData, findMatchingLoginNetwork } from '../../../utils/login';
 import getNetwork, { getNetworksList } from '../../../utils/getNetwork';
 import { parseSearchParams } from '../../../utils/searchParams';
-
 import Icon from '../../toolbox/icon';
 import UserAccount from '../../topBar/accountMenu/userAccount';
 import networks from '../../../constants/networks';
 import styles from './signInHeader.css';
 import formStyles from '../../send/form/form.css';
 import Dropdown from '../../toolbox/dropdown/dropdown';
 import Spinner from '../../spinner/spinner';
+import keyCodes from '../../../constants/keyCodes';
 import svg from '../../../utils/svgIcons';
 
 class Header extends React.Component {
   // eslint-disable-next-line max-statements
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
     const { liskCoreUrl } = getAutoLogInData();
     let loginNetwork = findMatchingLoginNetwork();
     let address = '';
 
     if (loginNetwork) loginNetwork = loginNetwork.slice(-1).shift();
     if (!loginNetwork) {
       loginNetwork = liskCoreUrl ? networks.customNode : networks.default;
-      address = liskCoreUrl || '';
+      address = liskCoreUrl || props.address;
     }
+
     this.state = {
       address,
       showDropdown: false,
       showSettingDrowdown: false,
       network: loginNetwork.code,
       isFirstTime: true,
     };
+
     this.toggleDropdown = this.toggleDropdown.bind(this);
     this.handleSettingsToggle = this.handleSettingsToggle.bind(this);
+    this.onConnectToCustomNode = this.onConnectToCustomNode.bind(this);
+  }
+
+  shouldComponentUpdate(nextProps) {
+    const { address } = this.props;
+    if (address !== nextProps.address) {
+      this.setState({ address: nextProps.address });
+      return false;
```

Comment:
Why component shouldn't update in this case? I think it should always update, so I think it should use `componentDidUpdate` instead of `shouldComponentUpdate`. Maybe I'm missing something...

Reply:
you are totally right @slaweet I was using the incorrect lifecycle, I just made the change and is better now